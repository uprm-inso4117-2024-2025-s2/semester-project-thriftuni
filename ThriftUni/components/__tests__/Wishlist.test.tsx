import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
// ⚠️ Adjust the import path if your Wishlist component lives elsewhere
import WishlistScreen from '../../app/(tabs)/WishlistPage'; // Adjust the import path as necessary

/**
 * Mutation‑oriented test‑suite for the Wishlist feature.
 * Mirrors the style of the previous Main‑Page mutation tests you provided:
 *   – Paired regex + exact string assertions
 *   – Placeholder queries
 *   – Button‑label existence checks
 *   – Pressability sanity check
 *   – Snapshot catch‑all
 * Update any text/placeholder strings to match your actual component labels.
 */

describe('Wishlist Screen', () => {
  // Unit Test: Header Text
  it('renders the header correctly', () => {
    const { getByText } = render(<WishlistScreen />);
    expect(getByText(/Wishlist/i)).toBeTruthy();   // Mutation‑safe (case‑insensitive)
    expect(getByText('My Wishlist')).toBeTruthy(); // Will fail if header text mutates
  });

  // Unit Test: Button Labels (mutation detection for renames/removals)
    // Unit Test: Key Controls (labels present)
    it('renders key controls with correct labels', () => {
        const { getByText } = render(<WishlistScreen />);
        // Header buttons
        expect(getByText('←')).toBeTruthy();   // Back arrow
        expect(getByText('Sort')).toBeTruthy();
        // Default Picker labels ensure filter/sort pickers are rendered
      });
    

  // Mutation Test: Button Interactivity
  it('sort button is pressable', () => {
    const { getByText } = render(<WishlistScreen />);
    const sortButton = getByText('Sort');
    fireEvent.press(sortButton); // Ensures pressability
    expect(sortButton).toBeTruthy();
  });

  // Mutation Test: UI Snapshot
  it('matches UI snapshot (layout, icons, styles)', () => {
    const tree = render(<WishlistScreen />).toJSON();
    expect(tree).toMatchSnapshot(); // Detects layout/style/icon mutations
  });
});
