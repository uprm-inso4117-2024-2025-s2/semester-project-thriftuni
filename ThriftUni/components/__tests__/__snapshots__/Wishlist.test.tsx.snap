// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wishlist Screen matches UI snapshot (layout, icons, styles) 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 24,
          }
        }
      >
        ‚Üê
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      My Wishlist
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "blue",
            "fontSize": 16,
          }
        }
      >
        Sort
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "black",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNCPicker
        items={
          [
            {
              "label": "Filter: All",
              "testID": undefined,
              "textColor": undefined,
              "value": "all",
            },
            {
              "label": "Available",
              "testID": undefined,
              "textColor": undefined,
              "value": "available",
            },
            {
              "label": "Unavailable",
              "testID": undefined,
              "textColor": undefined,
              "value": "unavailable",
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNCPicker
        items={
          [
            {
              "label": "Sort: None",
              "testID": undefined,
              "textColor": undefined,
              "value": "none",
            },
            {
              "label": "Price: Low to High",
              "testID": undefined,
              "textColor": undefined,
              "value": "asc",
            },
            {
              "label": "Price: High to Low",
              "testID": undefined,
              "textColor": undefined,
              "value": "desc",
            },
            {
              "label": "Name: A-Z",
              "testID": undefined,
              "textColor": undefined,
              "value": "asc",
            },
            {
              "label": "Name: Z-A",
              "testID": undefined,
              "textColor": undefined,
              "value": "desc",
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <ActivityIndicator
    color="blue"
    size="large"
  />
</View>
`;
